st2common/st2common/models/db/__init__.py

```diff
@@ -441,28 +445,35 @@ def distinct(self, *args, **kwargs):
         return result
 
     def aggregate(self, *args, **kwargs):
-        return self.model.objects(**kwargs)._collection.aggregate(*args, **kwargs)
+        with Timer(key='mongo.aggregate.' + self.model_name):
```

Comment:
That's fine, for a bigger picture, I guess, but it won't give us that much insight since operations can and do vary vastly based on the filters and kwargs used.

Reply:
Right, I think that's when your profiler would help. This is just to get a sense of cost of all the db operations and how many we are doing.

Reply:
As you pointed out, I removed the DB instrumentation since we already have the profiler.